home/tests.py

```diff
@@ -205,3 +208,149 @@ def tags_response(self, client):
+            # User entered all the required fields
+            ({'title': "Question in Math", 'content': 'How much is it 1+1?',
+              'subject': 1, 'grade': Grade.GRADE7, 'sub-subject': 2}),
+            # User entered all the required fields and sub-subject
+            ({'title': "Question in Math", 'content': 'How much is it 1+1?',
+              'subject': 1, 'grade': Grade.GRADE7, 'book': 2}),
+            # User entered all the required fields and sub-subject and book
+            ({'title': "Question in Math", 'content': 'How much is it 1+1?',
+              'subject': 1, 'grade': Grade.GRADE7, 'book_page': 23}),
+            # User entered all the required fields and a valid book page
+        ])
+        @pytest.mark.django_db
+        def test_post_valid_question_with_client(self, client, valid_data):
+            latestInsertedQuestionID = Question.objects.latest('id').id
+            client.login(username='Lior', password='LiorLior')
+            client.post('/explore/new_question', data=valid_data)
+
+            assert Question.objects.latest('id').id != latestInsertedQuestionID
+
+        @pytest.mark.parametrize("invalid_data", [
+            ({'content': 'How much is it 1+1?', 'subject': 1, 'grade': Grade.GRADE7}),
+            # User entered all the required fields except title
+            ({'title': "Question in Math", 'subject': 1, 'grade': Grade.GRADE7}),
+            # User entered all the required fields except content
+            ({'title': "Question in Math", 'content': 'How much is it 1+1?', 'grade': Grade.GRADE7}),
+            # User entered all the required fields except subject
+            ({'title': "Question in Math", 'content': 'How much is it 1+1?', 'subject': 1}),
+            # User entered all the required fields except grade
+            ({'title': "Question in Math", 'content': 'How much is it 1+1?', 'subject': 1, 'book_page': -23}),
+            # User entered all the required fields and a negative book page
+            ({'title': "Question in Math", 'content': 'How much is it 1+1?', 'subject': 1, 'book_page': 43343}),
+            # User entered all the required fields and to high book page
+        ])
+        @pytest.mark.django_db
+        def test_post_invalid_question_with_client(self, client, invalid_data):
+            latestInsertedQuestionID = Question.objects.latest('id').id
+            client.login(username='Lior', password='LiorLior')
+            client.post('/explore/new_question', data=invalid_data)
+
+            assert Question.objects.latest('id').id == latestInsertedQuestionID
+
+        @pytest.mark.django_db
+        def test_uses_question_form(self, client):
+            response = client.get('/explore/new_question')
+
+            assert response.status_code == 200
+            assert isinstance(response.context['form'], type(QuestionForm))
+
+        @pytest.mark.django_db
+        def test_renders_add_question_template(self, client):
```

Comment:
it looks like this test can be merged with the one above it, I don't think there is a need to test the context and the template in separate tests.

Reply:
> it looks like this test can be merged with the one above it, I don't think there is a need to test the context and the template in separate tests.

Ok, I will merge them